integration-tests/test/sync-verifier.spec.ts

```diff
@@ -0,0 +1,149 @@
+import chai, { expect } from 'chai'
+import { Wallet, BigNumber, Contract, ContractFactory, providers } from 'ethers'
+import { ethers } from 'hardhat'
+import { injectL2Context } from '@eth-optimism/core-utils'
+
+import { sleep, l2Provider } from './shared/utils'
+import { OptimismEnv } from './shared/env'
+import { DockerComposeNetwork } from './shared/docker-compose'
+
+describe('Syncing a verifier', () => {
+  let env: OptimismEnv
+  let wallet: Wallet
+  let verifier: DockerComposeNetwork
+
+  const provider = injectL2Context(l2Provider)
+
+  /** Helper functions **/
+
+  const waitForBatchSubmission = async (totalElementsBefore: BigNumber) => {
+    // Wait for batch submission to happen by watching the CTC
+    let totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    while (totalElementsBefore.eq(totalElementsAfter)) {
+      await sleep(500)
+      console.log(
+        `still equal`,
+        totalElementsBefore.toNumber(),
+        totalElementsAfter.toNumber()
+      )
+      totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    }
+    return totalElementsAfter
+  }
+
+  const startAndSyncVerifier = async (sequencerBlockNumber: number) => {
+    // Bring up new verifier
+    verifier = new DockerComposeNetwork(['verifier'])
+    await verifier.up({ commandOptions: ['--scale', 'verifier=1'] })
+
+    // Wait for verifier to be looping
+    let logs = await verifier.logs()
+    while (!logs.out.includes('Starting Sequencer Loop')) {
+      console.log('Retrieving more logs')
+      await sleep(500)
+      logs = await verifier.logs()
+    }
+
+    const verifierProvider = injectL2Context(
+      new providers.JsonRpcProvider('http://localhost:8547')
+    )
+    console.log(await verifierProvider.getBlock('latest'))
+
+    // Wait until verifier has caught up to the sequencer
+    let latestVerifierBlock = (await verifierProvider.getBlock('latest')) as any
+    while (latestVerifierBlock.number < sequencerBlockNumber) {
+      console.log('waiting for new verifier blocks')
+      await sleep(500)
+      latestVerifierBlock = (await verifierProvider.getBlock('latest')) as any
+    }
+
+    return latestVerifierBlock
+  }
+
+  before(async () => {
+    env = await OptimismEnv.new()
+    wallet = env.l2Wallet
+  })
+
+  describe('Basic transactions', () => {
+    const initialAmount = 1000
+    const tokenName = 'OVM Test'
+    const tokenDecimals = 8
+    const TokenSymbol = 'OVM'
+
+    let other: Wallet
+    let Factory__ERC20: ContractFactory
+    let ERC20: Contract
+
+    before(async () => {
+      const env = await OptimismEnv.new()
+      wallet = env.l2Wallet
+      other = Wallet.createRandom().connect(ethers.provider)
+      Factory__ERC20 = await ethers.getContractFactory('ERC20', wallet)
+    })
+
+    afterEach(async () => {
+      await verifier.stop('verifier')
+      await verifier.rm()
+    })
+
+    it('should sync dummy transaction', async () => {
```

Comment:
this test currently passes!

Reply:
This test halts for me at `waiting for new verifier blocks, current:  1 latest 2` when run with `it.only`

Reply:
@gakonst did you quit a previous test or had a verifier running locally before you started? i just remembered i ran into that before i removed the verifier image after every test; instead of restarting to sync the verifier just sits at the previous block